charms/docker/dockeropts.py

```diff
@@ -62,6 +62,21 @@ def add(self, key, value, strict=False):
             self.data[key] = None
         self.__save()
 
+    def pop(self, key):
+        '''
+        Completely remove a flag from the DockerOpts manager including any
+        associated values. Assuming the data is currently:
+        {'foo': ['bar', 'baz']}
+
+        d.pop('foo')
+        > {}
+
+        :params key:
+        '''
+
+        self.data.pop(key)
```

Comment:
Nope :) I don't want to return anything, this is strictly to whole-sale remove flags/values from the backing dict, as it causes problems when you've only removed the value and leave the flag behind in instances like "--bip". Without a value, it causes the daemon to fail cycle, and the flag manager appears to have not done its due diligence in cleaning up the value.

Thus the necessity for another method that fully removes keys, or adding an optional parameter to fully remove flags/values. I opted for the latter, as its becomes an explicit action then.